frost-core/src/frost/keys.rs

```diff
@@ -322,7 +323,13 @@ pub fn keygen_with_dealer<C: Ciphersuite, R: RngCore + CryptoRng>(
 
     let secret = SharedSecret::random(&mut rng);
     let group_public = VerifyingKey::from(&secret);
-    let secret_shares = generate_secret_shares(&secret, num_signers, threshold, rng)?;
+
+    let coefficients =
+        iter::repeat_with(|| <<C::Group as Group>::Field as Field>::random(&mut rng))
+            .take(threshold as usize - 1)
+            .collect();
```

Comment:
We've repeated this in tests, we may want to make it its own function

Reply:
Good idea, done in 4a7a9afd25eaeb9b215e58cc80605136a4608836